src/main/java/io/vertx/core/spi/loadbalancing/PreferenceAwareEndpointSelector.java

```diff
@@ -0,0 +1,16 @@
+package io.vertx.core.spi.loadbalancing;
+
+import java.util.List;
+
+public interface PreferenceAwareEndpointSelector extends EndpointSelectorCommons {
+
+  /**
+   * Select an endpoint among the provided list taking an attribute into account.
+   *
+   * @param endpoints the endpoint
+   * @param attribute the attribute TODO
+   * @return the selected index or {@code -1} if selection could not be achieved
+   */
+  int selectEndpoint(List<? extends Endpoint<?>> endpoints, String attribute);
```

Comment:
or should we allow passing this attr/preference to all policies? it can be confusing.